src/legacy/core_plugins/vis_type_vislib/public/plugin.ts

```diff
@@ -59,10 +59,10 @@ export interface VisTypeVislibPluginStartDependencies {
   visualizations: VisualizationsStart;
 }
 
-type VisTypeVislibCoreSetup = CoreSetup<VisTypeVislibPluginStartDependencies>;
+type VisTypeVislibCoreSetup = CoreSetup<VisTypeVislibPluginStartDependencies, void>;
 
 /** @internal */
-export class VisTypeVislibPlugin implements Plugin<Promise<void>, void> {
+export class VisTypeVislibPlugin implements Plugin<void, void> {
```

Comment:
Unwrapped a few useless promise types as `setup(core: CoreSetup, plugins: TPluginsSetup): TSetup | Promise<TSetup>` already handles it.